src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -79,65 +72,14 @@ class GitVersionPlugin implements Plugin<Project> {
 
     @Memoized
     private Git gitRepo(Project project) {
-        File gitDir = GitCli.getRootGitDir(project.projectDir);
+        File gitDir = GitCli.getRootGitDir(project.projectDir)
         return Git.wrap(new FileRepository(gitDir))
     }
 
     @Memoized
     private String gitDescribe(Project project, String prefix) {
-        // verify that "git" command exists (throws exception if it does not)
-        GitCli.verifyGitCommandExists()
-
-        def runGitCmd = { String... commands ->
-            return GitCli.runGitCommand(project.projectDir, commands);
-        }
-
-        Git git = gitRepo(project)
-        try {
-            // back-compat: the JGit "describe" command throws an exception in repositories with no commits, so call it
-            // first to preserve this behavior in cases where this call would fail but native "git" call does not.
-            new DescribeCommand(git.getRepository()).call()
-
-            /*
-             * Mimick 'git describe --tags --always --first-parent --match=${prefix}*' by using rev-list to
-             * support versions of git < 1.8.4
-             */
-
-            // Get SHAs of all tags, we only need to search for these later on
-            Set<String> tagRefs = Sets.newHashSet()
-            for (String tag : getLines(runGitCmd("show-ref", "--tags", "-d"))) {
-                List<String> parts = WORD_SPLITTER.splitToList(tag)
-                Preconditions.checkArgument(parts.size() == 2, "Could not parse output of `git show-ref`: %s", parts)
-                tagRefs.add(parts.get(0))
-            }
-
-            List<String> revs = getLines(runGitCmd("rev-list", "--first-parent", "HEAD"))
-            for (int depth = 0; depth < revs.size(); depth++) {
-                String rev = revs.get(depth)
-                if (tagRefs.contains(rev)) {
-                    String exactTag = runGitCmd("describe", "--tags", "--exact-match", "--match=${prefix}*", rev)
-                    if (exactTag != "") {
-                        return depth == 0 ?
-                                exactTag : String.format("%s-%s-g%s", exactTag, depth, abbrevHash(revs.get(0)))
-                    }
-                }
-            }
-
-            // No tags found, so return commit hash of HEAD
-            return abbrevHash(runGitCmd("rev-parse", "HEAD"))
-        } catch (Throwable t) {
-            return null
-        }
-    }
-
-    @Memoized
-    private List<String> getLines(String s) {
-        return LINE_SPLITTER.splitToList(s)
-    }
-
-    @Memoized
-    private String abbrevHash(String s) {
-        return s.substring(0, SHA_ABBR_LENGTH)
+        // return new NativeGitDescribe(project.projectDir).describe(prefix)
```

Comment:
did you mean to keep this comment?

Reply:
It's really scary to break this plugin considering it's used everywhere, and I'm not expert on git/jgit to be sure I've handled all edge cases.
@iamdanfox and I were thinking of some safe approach:
- compute both
- if one returns null, use the other one
- if both commands succeed, check if output is same
- if it's not, raise an exception (so we can fix the jgit implementation)
- eventually drop native one, once we're confident we have the same behaviour